src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -540,6 +540,7 @@ ucs_status_t uct_tcp_sockcm_ep_send(uct_tcp_sockcm_ep_t *cep)
     }
 
     if (uct_tcp_sockcm_ep_send_skip_event(cep)) {
+        ucs_assert(!(cep->state & UCT_TCP_SOCKCM_EP_DISCONNECTING));
```

Comment:
@evgeny-leksikov isn't this a bit weird that we first check flags and call uct_tcp_sockcm_ep_resolve() and only then check if this even should be skipped?

Reply:
I do not think so, prior to send something we have to pack something

Reply:
That makes sense but i was referring to the skip check. Just thought first we check if we need to skip the event and then, if not, continue to packing and sending.

Reply:
to start packing we have to invoke CB before

Reply:
as discussed offline, a sending function handling a write event but checks if an event can be skipped after already handling this event, is confusing and can be resolved by renaming the functions or removing the skip condition as suggested here - https://github.com/openucx/ucx/pull/6668/files#r623366911 

can be done later.